app/src/org/commcare/android/tasks/ManageKeyRecordTask.java

```diff
@@ -275,24 +301,35 @@ protected boolean processSuccesfulRequest() {
         SqlStorage<UserKeyRecord> storage = app.getStorage(UserKeyRecord.class);
         try {
             //We successfully received and parsed out some key records! Let's update the db
-            for(UserKeyRecord record : keyRecords) {
-                //See if we already have a key record for this sandbox and user  (There should _definitely_ only be one if there is one)
+            for (UserKeyRecord record : keyRecords) {
+
+                // See if we already have a key record for this sandbox and user
+                // (There should _definitely_ only be one if there is one)
                 UserKeyRecord existing;
+
                 try {
-                    existing = storage.getRecordForValues(new String[] {UserKeyRecord.META_SANDBOX_ID, UserKeyRecord.META_USERNAME} , new Object[] {record.getUuid(), record.getUsername()});
+                    existing = storage.getRecordForValues(
+                            new String[]{UserKeyRecord.META_SANDBOX_ID, UserKeyRecord.META_USERNAME},
+                            new Object[]{record.getUuid(), record.getUsername()});
                     
                     Logger.log(AndroidLogger.TYPE_MAINTENANCE, "Got new record for existing sandbox " + existing.getUuid() + " . Merging");
                     //So we have an existing record. Either we're updating our current record and we're updating the key details
                     //or our password has changed and we need to overwrite the existing key record. Either way, all
                     //we should need to do is merge the records.
                     
-                    UserKeyRecord ukr = new UserKeyRecord(record.getUsername(), record.getPasswordHash(), record.getEncryptedKey(), record.getValidFrom(), record.getValidTo(), record.getUuid(), existing.getType());
+                    UserKeyRecord ukr = new UserKeyRecord(
+                            record.getUsername(), record.getPasswordHash(),
+                            record.getEncryptedKey(), record.getWrappedPassword(),
+                            record.getValidFrom(), record.getValidTo(), record.getUuid(),
+                            existing.getType());
                     ukr.setID(existing.getID());
                     storage.write(ukr);
-                } catch(NoSuchElementException nsee) {
-                    //If there's no existing record, write this new one (we'll handle updating the status later)
+                } catch (NoSuchElementException nsee) {
+                    // If there's no existing record, write this new one (we'll handle updating the status later)
                     storage.write(record);
                 }
+
+                markOldRecordsInactive(storage, record.getUsername(), record.getUuid());

```

Comment:
I think this can be either "record" _or_ "ukr/existing" from above. In the non-catch block, "record" never receives a recordId (seems like that should be what determines whether the record isn't marked inactive?)